bento_aggregation_service/app.py

```diff
@@ -21,24 +24,56 @@
 from .search.handlers.datasets import DatasetsSearchHandler
 from .search.handlers.private_dataset import PrivateDatasetSearchHandler
 
+path_for_git = os.path.abspath(os.path.join(os.path.dirname(__file__), "../"))
+
+
+def before_first_request_func():
+    try:
+        subprocess.run(["git", "config", "--global", "--add", "safe.directory", str(path_for_git)])
+    except Exception as e:
+        except_name = type(e).__name__
+        print("Error in dev-mode retrieving git folder configuration", except_name)
 
+before_first_request_func()
 # noinspection PyAbstractClass,PyAttributeOutsideInit
+
 class ServiceInfoHandler(RequestHandler):
     async def get(self):
-        # Spec: https://github.com/ga4gh-discovery/ga4gh-service-info
-        self.write({
+        info_service = {
             "id": SERVICE_ID,
             "name": SERVICE_NAME,  # TODO: Should be globally unique?
             "type": SERVICE_TYPE,
             "description": "Aggregation service for a Bento platform node.",
+            "environment": "prod",
             "organization": {
                 "name": "C3G",
                 "url": "https://www.computationalgenomics.ca"
             },
             "contactUrl": "mailto:david.lougheed@mail.mcgill.ca",
             "version": bento_aggregation_service.__version__
-        })
+        }
+        if CHORD_DEBUG:
+            info = {
+                **info_service,
+                "environment": "dev",
+                }
+            try:
+                res_tag = subprocess.check_output(["git", "describe", "--tags", "--abbrev=0"])
```

Comment:
subprocess is blocking and should not be used in async-mode services; see my service registry PR for an example of doing non-blocking subprocess calls: https://github.com/bento-platform/bento_service_registry/pull/6/files#diff-2d9f7cd1baaf863258d8e642f3b8af349ba9fcf0ffbe458e66eb3d4cbd254e30R161-R168